### PR TITLE
LPS-156561, LPS-156563,  LPS-156330, LPS-156343, LPS-156327

### PR DESCRIPTION
**Pull request by thaiscabrals**
https://issues.liferay.com/browse/LPS-156330
https://issues.liferay.com/browse/LPS-156561
https://issues.liferay.com/browse/LPS-156563
https://issues.liferay.com/browse/LPS-156343
https://issues.liferay.com/browse/LPS-156327

### CODE REVIEW

**Comment by thaiscabrals** [comment]
ci:test:sf

**Comment by evanilson-santana** [comment]
Hi, @thaiscabrals I will review your PR.
cc: @timpak

**Comment by thaiscabrals** [comment]
ci:test:sf

**Comment by thaiscabrals** [comment]
ci:test:sf

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 1]
```diff
@@ -139,6 +139,41 @@ definition {
 			actionName = "Action Name");
 	}
 
+	@description = "LPS-145665. Verify that you can edit the Action name"
+	@priority = "4"
+	test CanEditActionName {
+		ObjectAdmin.addObjectViaAPI(
+			labelName = "Custom Object 145",
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:219]: It should be `LPS-145665 - Verify` following the convention in other tests

File: modules/apps/object/object-test/src/testFunctional/macros/ObjectAdmin.macro [hunk 2]
```diff
@@ -52,6 +53,11 @@ definition {
 			actionName = "${actionName}",
 			active = "${active}",
 			click = "save",
+			inputCondition = "${inputCondition}",
+			inputExpression = "${inputExpression}",
+			inputMethod = "${inputMethod}",
+			inputValue = "${inputValue}",
+			objectLabel = "${objectLabel}",
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/macros/ObjectAdmin.macro:60]: What happens to other tests that are using this macro but don't have these values set? Will they still pass?

File: modules/apps/object/object-test/src/testFunctional/macros/ObjectAdmin.macro [hunk 3]
```diff
@@ -837,21 +848,29 @@ definition {
...
-		}
+				ClickNoError.clickNoMouseOver(
+					key_objectLabel = "${objectLabel}",
+					locator1 = "ObjectAdmin#CHOOSE_OBJECT_LABEL");
 
-		if (isSet(inputAction)) {
-			if ("${inputActionActive}" == "true") {
 				Click(
-					locator1 = "ObjectAdmin#INPUT_METHOD",
-					value1 = "${inputMethod}");
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/macros/ObjectAdmin.macro]: What happens to tests that are setting `onAction` and `inputAction`?

File: modules/apps/object/object-test/src/testFunctional/paths/ObjectAdmin.path [hunk 4]
```diff
@@ -512,7 +512,17 @@
 	</tr>
 	<tr>
 		<td>INPUT_METHOD</td>
-		<td>//input[contains(@class, 'custom-control-input') and contains(@type,'checkbox')]/..//span[contains(text(),'${inputMethod}')]</td>
+		<td>//input[contains(@class, 'custom-control-input') and contains(@type, 'checkbox')]//..//span[contains(text(), '${key_inputMethod}')]</td>
+		<td></td>
+	</tr>
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/paths/ObjectAdmin.path:517]: If you need to fix your path, you should fix this in the commit that added it. Fixing it in another commit is messy.

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase [hunk 5]
```diff
@@ -21,6 +21,79 @@ definition {
...
+				active = "false",
+				viewEntry = "Custom Action");
+
+			ObjectAdmin.goToCustomObject(objectName = "CustomObject");
+
+			ObjectPortlet.addSingleFieldEntryViaUI(entry = "Custom Entry 2");
+
+			ObjectPortlet.assertEntryNotPresent(entry = "Custom Entry 2");
+		}
+	}
```
Comment (timpak) [inline modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:95]: I'm confused, why wouldn't the entry be added if the Action is disabled? Shouldn't it be created or am I missing something?

**Review by timpak — CHANGES_REQUESTED** [review]
Hi @thaiscabrals, please see my comments for the required fixes. Also, the final commit is missing the ticket from the commit message, which is why it's failing SF.

**Comment by timpak** [comment]
@evanilson-santana FYI. You can take a look at my comments to see what fixes are needed. Also, some of the changes I'm not sure how they will affect other tests. You may need to test the affected tests as well to make sure they still pass.

**Comment by timpak** [comment]
Also, @evanilson-santana can you assign the tickets to yourself? You can move them into `In Progress` if you need to do some more work on them

**Comment by evanilson-santana** [comment]
ci:reopen

**Comment by evanilson-santana** [comment]
ci:close

File: modules/apps/object/object-test/src/testFunctional/macros/ObjectAdmin.macro [hunk 6]
```diff
@@ -30,6 +30,21 @@ definition {
 		}
 	}
 
+	macro addObjectActionEntryViaUi {
+		ObjectAdmin.gotoAddObjectAction();
+
```
[the remark this answers is not in the document]
Comment (evanilson-santana) [reply modules/apps/object/object-test/src/testFunctional/macros/ObjectAdmin.macro:35]: Fixed

File: modules/apps/object/object-test/src/testFunctional/macros/ObjectAdmin.macro [hunk 7]
```diff
@@ -446,7 +461,6 @@ definition {
 	macro assertObjectActions {
 		AssertElementPresent(
 			key_actionActive = "${actionActive}",
-			key_actionName = "${actionName}",
 			locator1 = "ObjectAdmin#VIEW_ACTIONS_TAB");
 	}
```
[the remark this answers is not in the document]
Comment (evanilson-santana) [reply modules/apps/object/object-test/src/testFunctional/macros/ObjectAdmin.macro:471]: Fixed

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase (hunk 1 above)
Comment (evanilson-santana) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:219]: Fixed

File: modules/apps/object/object-test/src/testFunctional/macros/ObjectAdmin.macro (hunk 2 above)
Comment (evanilson-santana) [reply modules/apps/object/object-test/src/testFunctional/macros/ObjectAdmin.macro:60]: Fixed

File: modules/apps/object/object-test/src/testFunctional/macros/ObjectAdmin.macro (hunk 3 above)
Comment (evanilson-santana) [reply modules/apps/object/object-test/src/testFunctional/macros/ObjectAdmin.macro]: There is no test that set this

File: modules/apps/object/object-test/src/testFunctional/paths/ObjectAdmin.path (hunk 4 above)
Comment (evanilson-santana) [reply modules/apps/object/object-test/src/testFunctional/paths/ObjectAdmin.path:517]: Fixed

File: modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase (hunk 5 above)
Comment (evanilson-santana) [reply modules/apps/object/object-test/src/testFunctional/tests/ObjectActions.testcase:95]: Fixed

File: modules/apps/object/object-test/src/testFunctional/macros/ObjectAdmin.macro (hunk 2 above)
Comment (evanilson-santana) [reply modules/apps/object/object-test/src/testFunctional/macros/ObjectAdmin.macro:60]: What it seems to me is that `inputCondition` was being used as a variable to define a conditional in the macro. I modified it.

**Comment by evanilson-santana** [comment]
New PR [here](https://github.com/timpak/liferay-portal/pull/309)

**Pull request closed without merging**